src/sentry/static/sentry/app/views/organizationGroupDetails/groupSimilarIssues/similarTraceID/list.tsx

```diff
@@ -0,0 +1,198 @@
+      this.setState({isLoading: false, hasError: true});
+    }
+  };
+
+  // this little hack is necessary until we factored the groupStore or the EventOrGroupHeader component
+  // the goal of this function is to insert the properties eventID and groupID, so then the link rendered
+  // in the EventOrGroupHeader component will always have the structure '/organization/:orgSlug/issues/:groupId/event/:eventId/',
+  // providing a smooth navigation between issues with the same trace ID
+  convertGroupsIntoEventFormat = (groups: Array<Group>) => {
+    const {issues} = this.props;
+
+    return groups
+      .map(group => {
+        // the issue must always be found
+        const foundIssue = issues.find(issue => group.id === String(issue['issue.id']));
+        if (foundIssue) {
+          // the eventID is the reason why we need to use the DiscoverQuery component.
+          // At the moment the /issues/ endpoint above doesn't return this information
+          return {...group, eventID: foundIssue.id, groupID: group.id};
+        }
+        return undefined;
+      })
+      .filter(event => !!event) as Array<CustomGroup>;
+  };
+
+  handleCursorChange(cursor: string, path: string, query: Query, pageDiff: number) {
+    browserHistory.push({
+      pathname: path,
+      query: {...query, cursor: pageDiff <= 0 ? undefined : cursor},
+    });
+  }
+
+  handleRetry = () => {
+    this.getGroups();
+  };
+
+  renderContent = () => {
+    const {issues, orgSlug, period, traceID} = this.props;
+
+    if (!issues.length) {
+      return (
+        <EmptyStateWarning small withIcon={false}>
+          {tct(
+            'No issues with the same trace ID [traceID] were found in the period between [start] and [end]',
+            {
+              traceID,
+              start: <DateTime date={period.start} timeAndDate />,
+              end: <DateTime date={period.start} timeAndDate />,
+            }
+          )}
+        </EmptyStateWarning>
+      );
+    }
+
+    return issues.map(issue => (
+      <StreamGroup
+        key={issue.id}
+        id={String(issue['issue.id'])}
+        orgId={orgSlug}
+        canSelect={false}
+        withChart={false}
+      />
+    ));
+  };
+
+  render() {
+    const {pageLinks, traceID} = this.props;
+    const {isLoading, hasError} = this.state;
+
+    if (isLoading) {
+      return <LoadingIndicator />;
+    }
+
+    if (hasError) {
+      return (
+        <LoadingError
+          message={tct(
+            'An error occurred while fetching issues with the trace ID [traceID]',
+            {
+              traceID,
+            }
+          )}
+          onRetry={this.handleRetry}
+        />
+      );
+    }
+
+    return (
+      <React.Fragment>
+        <StyledPanel>
+          <GroupListHeader withChart={false} />
+          <PanelBody>{this.renderContent()}</PanelBody>
+        </StyledPanel>
+        <StyledPagination pageLinks={pageLinks} onCursor={this.handleCursorChange} />
```

Comment:
Would this make more sense to have in `<Body>`, since we paginating based on discover query.

Reply:
it would, but as I am displaying an empty state component or a list according to the response of a request.I prefer to leave it as it is..